pyfr/integrators/dual/pseudo/multip.py

```diff
@@ -288,6 +297,7 @@ def pseudo_advance(self, tcurr):
         self.tcurr = tcurr
 
         for i in range(self._maxniters):
+            csteps = tuple(self.wround(cstepf) for cstepf in cstepsf)
```

Comment:
Use choices [np.floor(c), np.ceil(c)] being the two choices with probabilities of [c % 1 and 1 - c % 1]